File: 2013-06-07/python/exercise6.py
## prendendo come riferimento: http://en.wikipedia.org/wiki/Wavefront_.obj_file

from pyplasm import *
import numpy

def lar_to_obj(modello):

	V = modello[0];
	FV = modello[1];
	s = "List of Vertices \n";

	for i in numpy.arange(0,len(V),1)

		if (V[i][3] == None): s += "v "+str(V[i][0])+" "+str(V[i][1])+ " " + str(V[i][2]) + " 1.0\n"
			
		else: s += "v "+str(V[i][0])+" "+str(V[i][1])+" "+str(V[i][2])+ " " + str(V[i][3])+"\n"
				
		
	s += "\n Face Definitions \n"
		for i in numpy.arange(0,len(FV),1):
			s += "f "+str(FV[i][0])+" "+str(FV[i][1])+" "+str(FV[i][2])+" "+str(FV[i][3])+"\n"
		

return(s)




l = lar_to_obj([[[0,0,3],[1,2,3]],[[5,4,6,8],[4,3,2,9]]]);

print(l)

File: 2013-06-07/javascript/exercise6.js
//prendendo come riferimento: http://en.wikipedia.org/wiki/Wavefront_.obj_file


var lar_to_obj = function(modello){

	var V = modello[0];

	var FV = modello[1];

	var s = "List of Vertices \n";

		for (var i = 0; i < V.length; i++){

			if (V[i][3] == null){
				s += "v "+V[i][0]+" "+V[i][1]+ " " + V[i][2] + " 1.0\n";}
			
			else {s += "v "+V[i][0]+" "+V[i][1]+" "+V[i][2]+ " " + V[i][3]+"\n"; }
				
		};
	s += "\n Face Definitions \n";
		for (var i = 0; i < FV.length; i++){
			s += "f "+FV[i][0]+" "+FV[i][1]+" "+FV[i][2]+" "+FV[i][3]+"\n";
		};

	return(s);
}



var l = lar_to_obj([[[0,0,3],[1,2,3]],[[5,4,6,8],[4,3,2,9]]]);

l

